### DISCUSSION TITLE
Virtual keyboard for password and text backup

### DISCUSSION THREAD

**Discussion by MiKeF05** [iotaledger/firefly#1571] [Features / Enhancements]
## Description

I thought about the input for the text backup and password of the stronghold file. Isn´t it less safe to enter these things directly with the keyboard? If the computer has a malware with a keylogger for example all things could be saved.

Maybe it is possible to use a virtual keyboard like in the image above where the user can enter the password of the stronghold file and each word of the text backup via this keyboard? For best safety the position of each letter and special character will be at a different position in the keyboard layout when it is loaded.

Wallet text backup

![Wallet Backup](https://user-images.githubusercontent.com/71659397/113562718-e2902600-9606-11eb-91ab-fcd5838a6a7b.PNG)

Virtual keyboard

![Virtual Keyboard](https://user-images.githubusercontent.com/71659397/113562744-f045ab80-9606-11eb-9bd3-b87cdaa3a58e.PNG)

## Motivation

To improve the safety of the wallet

## Requirements

Write a list of what you want this feature to do.

1. Enter the password for the stronghold file via keyboard layout
2. Enter the text backup of the wallet via keyboard layout

## Are you planning to do it yourself in a pull request?

No

**Comment by Phyloiota** [comment]
you can already use any kind of on screen keyboard now - i tried with windows on screen and also kaspersky on screen keyboard, both work for me. But yes a directly integrated one in firefly would also be a cool feature

**Comment by kajover** [comment]
+1
It would also improve security when the key placements are changing every time (not QWERTY)
